client/src/common/components/GridList.jsx

```diff
@@ -1,71 +1,105 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
+import { withRouter } from 'react-router-dom';
 import _map from 'lodash/map';
 import _isEmpty from 'lodash/isEmpty';
+import { connect } from 'react-redux';
 
 import CardItem from 'common/components/CardItem';
 import MessageBox from 'common/components/MessageBox';
 import { MessageType } from 'common/constants/enums';
 import CardPlus from 'common/components/CardPlus';
+import { manageFavourites } from 'modules/list/model/actions';
 
-const GridList = ({
-  color,
-  description,
-  icon,
-  items,
-  name,
-  onAddNew,
-  placeholder,
-  route
-}) => (
-  <div className="grid-list">
-    <h2 className="grid-list__heading">
-      {icon}
-      {name}
-    </h2>
-    <div className="grid-list__body">
-      <ul className="grid-list__list">
-        {onAddNew && (
-          <li className="grid-list__item">
-            <button
-              className="grid-list__button"
-              onClick={onAddNew}
-              type="button"
-            >
-              <CardPlus />
-            </button>
-          </li>
-        )}
-        {_map(items, item => (
-          <li className="grid-list__item" key={item._id}>
-            <Link to={`/${route}/${item._id}`}>
-              <CardItem
-                color={color}
-                description={item.description}
-                name={item.name}
-              />
-            </Link>
-          </li>
-        ))}
-      </ul>
-      {_isEmpty(items) && (
-        <MessageBox message={placeholder} type={MessageType.INFO} />
-      )}
-    </div>
-  </div>
-);
+export const GridListRoutes = Object.freeze({
+  COHORT: 'cohort',
+  LIST: 'list'
+});
+
+class GridList extends PureComponent {
+  handleFavClick = (itemId, isFavourite) => event => {
+    event.stopPropagation();
+    const { manageFavourites, route } = this.props;
+
+    route === GridListRoutes.LIST && manageFavourites(itemId, isFavourite);
```

Comment:
manageFavourites isn't required but you invoke it, maybe you should check it exist first?